streams/src/main/java/org/apache/kafka/streams/state/internals/StoreQueryUtils.java

```diff
@@ -62,4 +71,28 @@ public static void updatePosition(
             position.withComponent(meta.topic(), meta.partition(), meta.offset());
         }
     }
+
+    public static boolean isPermitted(
```

Comment:
Should we add this as a method in `Position`?

Reply:
I was hesitant to add anything extra to the Position class, but there's no particular reason for it.